pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +27,123 @@ import (
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:scope=Namespaced
 type StorageCohort struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the behavior of a cohort.
+	// +optional
+	Spec StorageCohortSpec `json:"spec,omitempty"`
+
+	// Most recently observed status of the cohort.
+	// Populated by the cohort operator or cohort manager.
+	// +optional
+	Status StorageCohortStatus `json:"status,omitempty"`
+}
+
+// StorageCohortSpec describes the attributes that a cohort is created with.
+type StorageCohortSpec struct {
+	// NodeSelector is used to specify the cohort to be considered
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+
+	// CohortManager stores the details of the cohort manager responsible for
+	// managing the cohort
+	// +optional
+	CohortManager CohortManagerDetails `json:"cohortManager,omitempty"`
+
+	// StorageProvisioner contains list of all provisioners responsible for
+	// the provisioning tasks for different storage solutions in the cohort
+	// +optional
+	StorageProvisioner []StorageProvisionerDetails `json:"storageProvisioner,omitempty"`
```

Comment:
Since we have decided to build a plug and play model for the pool and volume provisioners for individual cohort, this field plays an important role to let us know which provisioners are responsible for managing all data plane activities for a cohort. For reference you can check the detailed design document ->  https://docs.google.com/document/d/1Oj0rIuElrXwQgfBBq--e1nEiNE91lf2aLXXqtS5UWNQ/edit?usp=sharing

Reply:
Will the provisioner is unique to single cohort manager type? If yes, shouldn't this be a plain list of strings uniquely identifying the provisioner? How does storage type is relevant associated with provisioner relevant here?

Reply:
A storage provisioner is a plug-in which acts as a bridge between the Cohort (data-plane) and the SCP (control-plane). On one hand, the plug-in needs to understand the structure of the control-plane resources like Cohort, Pool and Volumes CRs. On the other, it has to translate the specs into actual Cohort-specific provisioning operations. A Cohort can support multiple types of storage, so a provisioner is identified by the type of storage (NOT cohort). The "type" in the storage provisioner struct matches with the type in pool spec.

Reply:
Just writing down the conclusion here - storageType is supposed to be used with pool definition, where individual provisioner will create pool for a given storageType with assumption that there will not be one to many mapping of storageType with provisioner. Since a storagetype will always be used by single provisioner, it's redundant and we may simplify it with single string (called provisioner) which will be used in pool definition and understood by each provisioner controller.